gcc/rust/checks/errors/borrowck/rust-borrow-checker.cc

```diff
@@ -155,19 +155,29 @@ BorrowChecker::go (HIR::Crate &crate)
       auto result
 	= Polonius::polonius_run (facts.freeze (), rust_be_debug_p ());
 
-      if (result.loan_errors)
+      // convert to std::vector variation for easier navigation
+      auto loan_errors = make_vector (result.loan_errors);
+      auto move_errors = make_vector (result.move_errors);
+      auto subset_errors = make_vector (result.subset_errors);
+
+      // free allocated data
+      delete result.loan_errors;
+      delete result.move_errors;
+      delete result.subset_errors;
```

Comment:
we'll do that in a later PR but I believe we can do away with this and simply use `std::move` to take the result's vectors and simplify the code. not necessary for now